### DISCUSSION TITLE
Publish Lambda Layer ARN publicly across all regions instead of SAR App

### DISCUSSION THREAD

**Discussion by ecolban** [aws-powertools/powertools-lambda-python#329] [Q&A] [answered]
**What were you trying to accomplish?**

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Environment

* **Powertools version used**:
* **Packaging format (Layers, PyPi)**:
* **AWS Lambda function runtime:**
* **Debugging logs**

> [How to enable debug mode](https://awslabs.github.io/aws-lambda-powertools-python/#debug-mode)**

```python
# paste logs here
```

**Comment by mwarkentin** [comment]
Is this actually an issue? You can install the app from any region while referencing eu-west-1.

**Comment by ecolban** [comment]
It is not a bug, but a desired feature that is missing. 

I can install it, but when deploying a lambda function, I cannot reference the layer by its ARN. Lambda requires the layer to be in the same region. If I'm not using SAM, the alternative is to pip install it locally and build a layer from the installed packages. It would be very nice if there were an ARN for each region. I also had problems finding the version.

**Comment by Nr18** [comment]
@ecolban do you mind adding that context to the description of the issue? That would make it easier to understand the request that you have. In general, an issue that is clear of what you expect and what you considered etc is more likely to get resolved more quickly.

**Comment by heitorlessa** [comment]
Hey @ecolban, if I'm understanding you correctly, **you want AWS to publish Lambda Layer ARNs for all AWS regions so that you can consume them directly** instead of deploying a SAR App - **Did I get this right?**

> **For example**: `arn:aws:lambda:<Region>:<AccountId>:layer:aws-lambda-powertools:<Version>`

If that's the case, we won't be able to do that at this stage until Lambda Layers support Aliases or any mechanism in which we could honour semantic versioning to match what's on PyPi - Please reach out to your AWS representative for this feature, or AWS Developer Advocates using `#awswishlist` on Twitter.

That is why we chose SAR as a distribution mechanism because it supports semantic versioning and we can match SAR App with what's on our source of truth (PyPi). The SAR App is distributed globally despite the region prefix, and it deploys a Lambda Layer in the region of your choosing, which you can then [ref in your favourite deployment framework](https://awslabs.github.io/aws-lambda-powertools-python/#lambda-layer).

**PS**: I've changed from bug to feature request

Thanks.

**Comment by ecolban** [comment]
Yes, I agree it should be a feature request and not a bug. Note that I found a layer ARN [here](https://github.com/keithrozario/Klayers/blob/master/deployments/python3.8/arns), but I would have preferred it being provided by you.

**Reply by heitorlessa** [reply]
Quick update on this to let you know @ecolban that we're discussing it internally on ways we could make this less burdensome to maintain while providing a better experience for Layers consumption too.

I'll update it once we have anything to share here on the decision

**Reply by heitorlessa** [reply]
Update - @am29d is working on this for all regions.

**Reply by heitorlessa** [reply]
This is now on the public roadmap: https://github.com/awslabs/aws-lambda-powertools-roadmap/projects/1

**Comment by heitorlessa** [comment, marked as answer]
This is now finally available: https://awslabs.github.io/aws-lambda-powertools-python/latest/#public-arns

We're deploying across all regions now and should take some time for it to be available. SAR will continue to be available due to support for semantic versions that's not supported in Lambda Layers.
